x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -343,6 +344,19 @@ export type ThresholdNormalized = t.TypeOf<typeof thresholdNormalized>;
 export const thresholdNormalizedOrUndefined = t.union([thresholdNormalized, t.undefined]);
 export type ThresholdNormalizedOrUndefined = t.TypeOf<typeof thresholdNormalizedOrUndefined>;
 
+// New terms rule type currently only supports a single term, but should support more in the future
+export const newTermsFields = LimitedSizeArray({ codec: t.string, minSize: 1, maxSize: 1 });
+export type NewTermsFields = t.TypeOf<typeof newTermsFields>;
+
+export const newTermsFieldsOrUndefined = t.union([newTermsFields, t.undefined]);
+export type NewTermsFieldsOrUndefined = t.TypeOf<typeof newTermsFieldsOrUndefined>;
```

Comment:
Here's a comment about using type aliases and switching from this `OrUndefined` pattern: https://github.com/elastic/kibana/pull/132247#pullrequestreview-982376475

Also, it seems like the `OrUndefined` types introduced here are not used. I'd change it to:

```ts
// New terms rule type currently only supports a single term, but should support more in the future
export const NewTermsFields = LimitedSizeArray({ codec: t.string, minSize: 1, maxSize: 1 });
export type NewTermsFields = t.TypeOf<typeof NewTermsFields>;

export type HistoryWindowStart = t.TypeOf<typeof HistoryWindowStart>;
export const HistoryWindowStart = t.string;
```

I'd also consider using `NonEmptyString` instead of `t.string` wherever it makes sense.